File: src/co/soundwave/repository/artist/ArtistDAO.kt
package co.soundwave.repository.artist

import co.soundwave.models.Artist
import co.soundwave.repository.BaseDAO

class ArtistDAO : BaseDAO<ArtistSQL, Artist>() {
    override val sql: ArtistSQL by lazy { ArtistSQL() }
    fun getInsertQuery(obj: Artist): String =
        sql.insertArtist(obj.name, obj.nick, obj.email, obj.password)
}

File: repository/artist/ArtistDAO.py
from models import Artist as art
from repository import BaseDAO as bdao
from repository.artist import ArtistSQL as artsql


class ArtistDAO(bdao.BaseDAO):
    def __init__(self):
        super().__init__()
        self.sql = artsql.ArtistSQL()

    def get_insert_query(self, artist: art.Artist) -> str:
        return self.sql.insert_artist(artist.name, artist.nick, artist.photo, artist.email,
                                      artist.password)
